spyder/utils/tests/test_conda.py

```diff
@@ -65,8 +65,23 @@ def test_get_conda_root_prefix():
 
 @pytest.mark.skipif(not running_in_ci(), reason="Only meant for CIs")
 def test_find_conda():
+    # Standard test
     assert find_conda()
 
+    # Test with test environment
+    # Temporarily remove CONDA_EXE and MAMBA_EXE, if present
+    conda_exe = os.environ.pop('CONDA_EXE', None)
+    mamba_exe = os.environ.pop('MAMBA_EXE', None)
+
+    pyexec = get_conda_test_env()[1]
```

Comment:
Please move this line to be before removing the env vars to be sure `get_conda_test_env` is returning the right value. That's because it depends on `find_conda`.